src/groupByKey.ts

```diff
@@ -2,6 +2,20 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  items.map((item) => {
+    const x = String(item[key]);
```

Comment:
Let's use more meaningful names than `x` and `result`. It will help to make your code cleaner and more understandable.

Reply:
Thank you!